docs: update air-gapped docs

itaysk opened the pull request:
discussed here https://github.com/aquasecurity/trivy/issues/7029#issuecomment-2225913815

I did my best to restructure the docs in a way that makes sense, I have a few open comments:

1. not sure how authentication works with `--checks-bundle-repository` @simar7 
2. do we have a "manually populate cache" option for checks? @simar7 
3. inconsistency in naming between `--checks-bundle-repository` and `--skip-check-update`, not sure we want to fix this at this point.
4. `--offline-scan` isn't clearly explained, I presume this is about maven central? then the name is misleading (should be java specific). again, not sure we want to fix now but wanted to bring to attention. Also I thought this can be expanded to disable all the external dependencies (all the DBs) in one flag.

knqyf263 commented on docs/docs/advanced/air-gap.md:
```diff
@@ -1,142 +1,125 @@
+For Trivy DB, configure it in the [same way as for private images](../advanced/private-registries/index.md).
 
-    Download `javadb.tar.gz`:
+For Java DB, you need to run `docker login YOUR_REGISTRY`. Currently, specifying a username and password is not supported.
```
The original document was incorrect. The vulnerability DB and Java DB work the same, and it supports only `docker login` now.

nikpivkin replied:
@knqyf263 Why does Java DB only support authentication via docker login?  When an artifact is loaded, [this](https://github.com/aquasecurity/trivy/blob/main/pkg/oci/artifact.go#L73) method is called, which loads the image with authentication.

knqyf263 replied:
It's my bad. I was using an old laptop today and referring to an old commit.

itaysk replied:
so just to confirm, vuln db, java db, checks db all support the same authentication, and that is trivy's "private registry" options. right?

itaysk replied:
https://github.com/aquasecurity/trivy/pull/7160/commits/4695eebf7a876ff18c9a56f07e4ef4c141dc7e10

simar7 replied:
Yes

knqyf263 commented on docs/docs/advanced/air-gap.md:
```diff
@@ -1,142 +1,125 @@
-    Download `db.tar.gz`:
+- `--skip-db-update` to skip updating the main vulnerability database.
+- `--skip-java-db-update` to skip updating the Java vulnerability database.
+- `--offline-scan` to scan Java applications without issuing API requests.
```
We no longer need it and can delete it for now. But @DmitriyLewen suggested re-implementing sonatype API. We'll add it back after that.

knqyf263 replied:
Also, users suggested that `--offline` enables all needed flags for air-gapped environments.

itaysk replied:
https://github.com/aquasecurity/trivy/pull/7160/commits/4695eebf7a876ff18c9a56f07e4ef4c141dc7e10

knqyf263 reviewed (commented):
>--offline-scan isn't clearly explained, I presume this is about maven central? then the name is misleading (should be java specific). again, not sure we want to fix now but wanted to bring to attention. Also I thought this can be expanded to disable all the external dependencies (all the DBs) in one flag.

`--offline-scan` should affect all scanning, not limited to JAR scanning (To complicate things, JAR scanning was switched to Java DB, so this flag is not used for anything now.). When we implement internet access in the future, it should be disabled by `--offline-scan`. Also, I think misconfiguration scanning should respect this flag (downloading Terraform modules, etc.).

nikpivkin commented:
1. The checks bundle is distributed as an OCI artifacts and retrieved from the Artifact registry. Authentication methods are described [here](https://aquasecurity.github.io/trivy/v0.53/docs/advanced/private-registries/#credential).

itaysk commented on docs/docs/advanced/air-gap.md:
````diff
@@ -1,142 +1,125 @@
+```
 
-=== "Trivy"
+### Authentication
````
I agree, I also thought about the allowlist and private access not belonging to the airgap document, but also didn't want to leave them in a vulnerability specific document since they are shared to misconfig and maybe other things in the future.
WDYT about renaming the air-gap document to "advanced database scenarios"? then it makes more sense why the network requirements, authentication, and offline settings are there. 
It does sounds similar to the existing "configuration/db.md" doc that you linked, but I thought that the entire "configuration" section is purely CLI flags reference (and I thought it's auto-generated)

knqyf263 replied:
>WDYT about renaming the air-gap document to "advanced database scenarios"? then it makes more sense why the network requirements, authentication, and offline settings are there.

Yes, It makes more sense, but since we also have checks bundle, advanced "database" scenarios might be misleading.

itaysk replied:
https://github.com/aquasecurity/trivy/pull/7160/commits/4695eebf7a876ff18c9a56f07e4ef4c141dc7e10

itaysk commented:
> The checks bundle is distributed as an OCI artifacts and retrieved from the Artifact registry. Authentication methods are described [here](https://aquasecurity.github.io/trivy/v0.53/docs/advanced/private-registries/#credential).

@nikpivkin , @knqyf263 just [said](https://github.com/aquasecurity/trivy/pull/7160#discussion_r1677319315) that the vuln DBs support only docker login, they don't support user/pass. is it the same for checks bundle as well? or checks bundle does support user/pass unlike vuln scanner?

nikpivkin commented:
@itaysk supports login and password

simar7 commented on docs/docs/advanced/air-gap.md:
```diff
@@ -1,142 +1,128 @@
 
-[allowlist]: ../references/troubleshooting.md
-[oras]: https://oras.land/docs/installation
+### Java DB
+
+For Java DB the process is the same, except for the following:
+1. Image location is `ghcr.io/aquasecurity/trivy-java-db:1`
+2. Archive file name is `javadb.tar.gz`
+3. DB file name is `trivy-java.db`
```
Formatting needs a fix

<img width="858" alt="image" src="https://github.com/user-attachments/assets/fa70a072-de69-484d-aa8c-69636527e7a5">

simar7 commented on docs/docs/scanner/misconfiguration/check/builtin.md:
```diff
@@ -1,21 +1,22 @@
 
-For suggestions or issues regarding policy content, please open an issue under the [trivy-checks][trivy-checks] repository.
+## Checks Bundle
+When performing a misconfiguration scan, Trivy will automatically downloads the relevant Checks bundle. The bundle is cached locally and Trivy will reuse it for subsequent scans on the same machine. Trivy takes care of updating the cache automatically so normally can be oblivious to it.
```
```suggestion
When performing a misconfiguration scan, Trivy will automatically download the relevant Checks bundle. The bundle is cached locally and Trivy will reuse it for subsequent scans on the same machine. Trivy takes care of updating the cache automatically, so the user can normally can be oblivious to it.
```

itaysk commented:
@knqyf263 wdyt about removing the manual cache manipulation guide? it seems very internal and i think using the flag is better

knqyf263 commented:
@itaysk We discussed it in https://github.com/aquasecurity/trivy/issues/7029 and agreed not to add flags.

knqyf263 commented:
Or do you mean we should force users to upload DB to their OCI registry?

itaysk commented:
I meant the latter. not saying it's a big problem, just seems redundant and makeshift to me. ok i'll keep it, but then I want to confirm what is the procedure for misconfig checks (@simar7 @nikpivkin )?

itaysk commented:
Added another node about loading config from directory using `--config-check` flag. Also found many outdates references to this flag which I fixed in the same commit. FTR, also adding this to the list of inconsistencies (checks vs check) @simar7 can you please review

nikpivkin commented on docs/docs/scanner/misconfiguration/check/builtin.md:
```diff
@@ -1,21 +1,22 @@
-When misconfiguration detection is enabled, Trivy pulls the OPA bundle from GHCR as an OCI artifact and stores it in the cache.
-Those checks are then loaded into Trivy OPA engine and used for detecting misconfigurations.
-If Trivy is unable to pull down newer checks, it will use the embedded set of checks as a fallback. This is also the case in air-gap environments where `--skip-policy-update` might be passed.
+For CLI flags related to the database, please refer to [this page](../configuration/db.md).
```
Database?

simar7 commented on docs/docs/scanner/misconfiguration/index.md:
```diff
@@ -326,13 +326,13 @@ trivy config --misconfig-scanners=terraform,dockerfile .
-### Passing custom checks
-You can pass policy files or directories including your custom checks through `--policy` option.
+### Loading custom checks
+You can load policy files or directories including your custom checks through `--config-check` flag.
```
For some reason, GitHub has seemed to disable suggestions for me. But in this line it should be `load check files` instead of policy.

simar7 commented on docs/docs/scanner/misconfiguration/custom/data.md:
````diff
@@ -31,5 +31,5 @@ Then, you need to pass data paths through `--data` option.
 
 ```bash
-$ trivy conf --policy ./policy --data data --namespaces user ./configs
+$ trivy conf --config-check ./checks --data ./data --namespaces user ./configs
````
It is a nit but I like to use `trivy config` instead of `trivy conf` as it is more descriptive.

simar7 commented on docs/docs/scanner/misconfiguration/custom/index.md:
```diff
@@ -2,10 +2,10 @@
 ## Overview
 You can write custom checks in [Rego][rego].
-Once you finish writing custom checks, you can pass the policy files or the directory where those policies are stored with `--policy` option.
+Once you finish writing custom checks, you can pass the policy files or the directory where those checks are stored with --config-check` option.
```
```suggestion
Once you finish writing custom checks, you can pass the check files or the directory where those checks are stored with --config-check` option.
```

simar7 commented:
> I meant the latter. not saying it's a big problem, just seems redundant and makeshift to me. ok i'll keep it, but then I want to confirm what is the procedure for misconfig checks (@simar7 @nikpivkin )?

The process would be the same as trivy-db, the users have to navigate Trivy's cache to place the checks in the right place in case of an air gap solution.

Maybe we should document this cache structure in more detail as a whole as it is a common place for all scanners?

itaysk commented:
@knqyf263 I've addressed vex hub in air gapped env, let me know if it's ok

knqyf263 commented on docs/docs/advanced/air-gap.md:
```diff
@@ -1,142 +1,162 @@
+
+## Misconfigurations scanning
+
+Note that the misconfigurations database is also embedded in the Trivy binary (at build time), and will be used as a fallback if the external database is not available. This means that you can still scan for misconfigurations in an air-gapped environment using the Checks from the time of the Trivy release you are using.
```
Do we call the checks bundle "database"?

itaysk replied:
I have some thoughts about this, but don't want to delay this PR with them so I'll do change it

knqyf263 reviewed (commented):
LGTM. Left some comments.

knqyf263 commented on docs/docs/advanced/air-gap.md:
I have an idea about re-structuring this page, but we can do that in another PR.
